pkg/controller/installation/installer.go

```diff
@@ -162,23 +162,60 @@ func (i *Installer) patchService(
 	labelsToInject map[string]string,
 	ownerReference *metav1.OwnerReference,
 ) runtime.Object {
-	appName, ok := labelsToInject[shipperv1.AppLabel]
-	if !ok {
-		// If we reach this point, it is fine to panic() since there's nothing
-		// much to do if we don't have the application label.
-		panic(fmt.Sprintf("Programmer error, label %q should always be present", shipperv1.AppLabel))
+
+	requiredLabels := []string{shipperv1.AppLabel}
+	for _, label := range requiredLabels {
+		if _, ok := labelsToInject[label]; !ok {
+			// If we reach this point, it is fine to panic() since there's nothing
+			// much to do if we don't have the mandatory labels.
+			panic(fmt.Sprintf("Programmer error, label %q should always be present", label))
+		}
 	}
 
 	// Those are modified regardless.
 	s.OwnerReferences = []metav1.OwnerReference{*ownerReference}
 	s.Labels = mergeLabels(s.Labels, labelsToInject)
-	s.Spec.Selector[shipperv1.AppLabel] = appName
+
+	return s
+}
+
+func (i *Installer) modifyServiceSelector(
+	s *corev1.Service,
+) (runtime.Object, error) {
+
+	labels := s.Labels
+	appName, ok := labels[shipperv1.AppLabel]
+	if !ok {
+		return nil, controller.NewInvalidChartError(
+			fmt.Sprintf("A service object metadata is expected to contain %q label, none found",
+				shipperv1.HelmAppLabel))
+	}
 
 	// We are interested only in patching Services properly identified by our specific label
-	if lbValue, ok := s.Labels[shipperv1.LBLabel]; ok && lbValue == shipperv1.LBForProduction {
+	if lbValue, ok := labels[shipperv1.LBLabel]; ok && lbValue == shipperv1.LBForProduction {
 		s.Spec.Selector[shipperv1.PodTrafficStatusLabel] = shipperv1.Enabled
 	}
-	return s
+
+	if relName, ok := s.Spec.Selector[shipperv1.HelmReleaseLabel]; ok {
+		if _, ok := i.InstallationTarget.Labels[shipperv1.HelmWorkaroundLabel]; ok {
+			// This selector label is native to helm-bootstrapped charts.
+			// In order to make it work the shipper way, we remove the
+			// label and proceed normally. With one little twist: the user
+			// has to ask shipper to do it explicitly.
+			if relName == i.Release.Name {
+				delete(s.Spec.Selector, shipperv1.HelmReleaseLabel)
+			}
+		} else {
+			return nil, controller.NewInvalidChartError(
+				fmt.Sprintf("The chart contains %q label. This will break shipper"+
```

Comment:
I like this error, but maybe we can make it clearer? Something like:

```
fmt.Sprintf("The chart contains %q label. This will break shipper"+
 					" traffic shifting logic. Consider adding the workaround label %q: true to your Application object",
 					shipperv1.HelmReleaseLabel, shipperv1.HelmWorkaroundLabel))
```

Reply:
Fixed in the recent commit. Thank you!